cdc/puller/ddl_puller.go

```diff
@@ -42,6 +49,154 @@ const (
 	DDLPullerTableName = "DDL_PULLER"
 )
 
+// DDLJobPuller is used to pull ddl job from TiKV.
+type DDLJobPuller interface {
+	Run(ctx context.Context) error
+	Output() <-chan *model.DDLJobEntry
+}
+
+type ddlJobPullerImpl struct {
+	puller    Puller
+	kvStorage tidbkv.Storage
+	tableInfo *model.TableInfo
+	columnID  int64
+	outputCh  chan *model.DDLJobEntry
+}
+
+func (p *ddlJobPullerImpl) Run(ctx context.Context) error {
+	g, ctx := errgroup.WithContext(ctx)
+
+	g.Go(func() error {
+		return p.puller.Run(ctx)
+	})
+
+	rawDDLCh := memory.SortOutput(ctx, p.puller.Output())
```

Comment:
When is `rawDDLCh` closed? Is it ever closed? Please justify your handling of the channel.

Reply:
seems in `SortOutput`, the rawDDLCh is not closed. I just copy the other usage of `SortOutput`, they are total the same